Classes/Service/RecordService.php

```diff
@@ -54,7 +54,11 @@ public function getSingle($table, $fields, $uid)
     public function update($table, array $record)
     {
         $connection = $this->getDatabaseConnection();
-        return $connection->exec_UPDATEquery($table, "uid = '" . intval($record['uid']) . "'", $record);
+        if ($GLOBALS['BE_USER']->workspace > 0 && intval($record['t3ver_wsid']) != $GLOBALS['BE_USER']->workspace) {
```

Comment:
Couple of things: the WorkspaceAwareRecordService is the one that is supposed to handle record retrieval in places where we need to respect workspaces. And checks are required to check if the `BE_USER` global is initialised - which it isn't if no user is logged in.